components/locale/src/extensions/transform/fields.rs

```diff
@@ -0,0 +1,33 @@
+use std::ops::Deref;
+
+use super::Key;
+use super::Value;
+
+#[derive(Clone, PartialEq, Eq, Debug, Default, Hash, PartialOrd, Ord)]
+pub struct Fields(Box<[(Key, Value)]>);
+
+impl Fields {
+    pub fn from_vec_unchecked(input: Vec<(Key, Value)>) -> Self {
+        Self(input.into_boxed_slice())
+    }
+
+    pub fn contains(&self, key: &Key) -> bool {
```

Comment:
All std collection `contains` take `&T`, so for all `Struct(Box<[T]>)` I was able to just `Deref` the `&[T] and its `contains` does indeed take `&T`.

But for places where I use `binary_search_by_key` under the hood, I need to explicitly implement `contains` and `clippy` recommends passing by value rather than reference.

If I do that, I'll end up with:

```rust
let fields: transform::Fields = "".parse().unwrap();
let key: transform::Key = "".parse().unwrap();
fields.contains(key); // pass by value to a custom `contains`

let private: private::Private = "".parse().unwrap();
let key: private::Key = "".parse().unwrap();

private.contains(&key); // pass by reference to the derefed `slice::contains`
```
@Manishearth  - what do you recommend? consistency or clippy recommendation?

Reply:
you can use `Borrow<T>` for making `.contains()` work on both, check out how hashmap's contains works

Reply:
Ah, damn! Thank you!

Reply:
For the future reference: https://doc.rust-lang.org/src/std/collections/hash/map.rs.html#741-747